From: /C=EU/ST=EU/CN=Patrick McHardy/emailAddress=kaber@trash.net </C=EU/ST=EU/CN=Patrick McHardy/emailAddress=kaber@trash.net>
Date: Thu, 15 Nov 2007 13:06:58 +0000 (+0000)
Subject: PATCH - Fix for --random option in DNAT and REDIRECT
X-Git-Tag: iptables_1_4_0~42
X-Git-Url: https://git.netfilter.org/cgi-bin/gitweb.cgi?p=iptables.old-history.git;a=commitdiff_plain;h=35427b903ea3077ac84192a8f1ff9f653b852732

PATCH - Fix for --random option in DNAT and REDIRECT

The --random option produces "Unknown arg `--random'" errors with both the
DNAT and REDIRECT targets. Corrected by the attached patch.

Tom Eastep <teastep@shorewall.net>
---

diff --git a/extensions/libipt_DNAT.c b/extensions/libipt_DNAT.c
index eaed57b..7322a20 100644
--- a/extensions/libipt_DNAT.c
+++ b/extensions/libipt_DNAT.c
@@ -178,6 +178,7 @@ static int DNAT_parse(int c, char **argv, int invert, unsigned int *flags,
 			*flags |= IPT_DNAT_OPT_RANDOM;
 		} else
 			*flags |= IPT_DNAT_OPT_RANDOM;
+		return 1;
 	default:
 		return 0;
 	}
diff --git a/extensions/libipt_REDIRECT.c b/extensions/libipt_REDIRECT.c
index 077b6d8..f749dd1 100644
--- a/extensions/libipt_REDIRECT.c
+++ b/extensions/libipt_REDIRECT.c
@@ -23,7 +23,7 @@ IPTABLES_VERSION);
 
 static struct option opts[] = {
 	{ "to-ports", 1, 0, '1' },
-	{ "random", 1, 0, '2' },
+	{ "random", 0, 0, '2' },
 	{ 0 }
 };
 
